source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,187 @@
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine, in these cases you can remove the `hardware-configuration.nix` import.
+
+To be able to log in, you must uncomment the section that specifies the user "alice", or add the following lines to the configuration inside the curly bracket that contain the configuration specification:
+```nix
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ];
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+  };
+```
+
+Additionally, you need to specify a password for this user.
+An insecure solution is to specify an unencrypted plaintext password by adding the `initialPassword` option to the user configuration:
+```nix
+    initialPassword = "danger";
+```
+
+Note that this may pose a security risk.
+A more secure option is to encrypt a password using sha-512.
+To create an encrypted password (here using "danger" as an example), execute:
+```shell-session
+$ mkpasswd -m sha-512 danger
+```
+
+Copy the output of `mkpasswd` into the user configuration and use the `initialHashedPassword` option.
+It should look something like this:
+```nix
+    initialHashedPassword = "$6$YLpneQLpnPxXpo1Y$mTZg26KMjWIBqP1N98LzeANb5rfMcC5t7a7Khf/gTB/rPCT4t4x2EgJJZmXkRWcGVW6ZEDMulsjTsXxD7BLZZ/";
+```
+
+The complete `configuration.nix` file now looks like this:
+```nix
+{ config, pkgs, ... }:
+{
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+    initialHashedPassword = "$6$YLpneQLpnPxXpo1Y$mTZg26KMjWIBqP1N98LzeANb5rfMcC5t7a7Khf/gTB/rPCT4t4x2EgJJZmXkRWcGVW6ZEDMulsjTsXxD7BLZZ/";
+  };
+
+  system.stateVersion = "22.05";
+}
+```
+
+## Manual testing of a NixOS configuration
+
+One of the most powerful features in the Nix ecosystem is **the ability
+to provide a set of declarative NixOS configurations and use a simple
+Python interface** to interact with them using [QEMU](https://www.qemu.org/)
+as the backend.
+Those tests are widely used to ensure that NixOS works as intended, so in general they are called **NixOS tests**.
+They can be written and launched outside of NixOS, on any Linux machine (with
+[MacOS support coming soon](https://github.com/NixOS/nixpkgs/issues/108984)).
+Integration tests are reproducible due to the design properties of Nix,
+making them a valuable part of a Continuous Integration (CI) pipeline. [^nixdev]
+
+### Creating a QEMU based virtual machine using a configuration
+
+A virtual machine is created with the `nix-build` command.
```

Comment:
Generally I suggest splitting (even such short) units of meaning into paragraphs. For reference: https://www.plainlanguage.gov/guidelines/concise/write-short-paragraphs/

```suggestion
A virtual machine is created with the `nix-build` command.

```